src/dbnode/client/session.go

```diff
@@ -3476,61 +3526,117 @@ const enqueueChannelDefaultLen = 32768
 func newEnqueueChannel(m *streamFromPeersMetrics) enqueueChannel {
 	c := &enqueueCh{
 		peersMetadataCh: make(chan []receivedBlockMetadata, enqueueChannelDefaultLen),
-		closed:          0,
 		metrics:         m,
 	}
+
 	// Allocate the enqueue delayed fn just once
-	c.enqueueDelayedFn = func(peersMetadata []receivedBlockMetadata) {
+	c.enqueueDelayedFn = func(peersMetadata []receivedBlockMetadata) error {
+		c.Lock()
+		if c.closed {
+			c.Unlock()
+			return errEnqueueChIsClosed
+		}
+		c.sending++
```

Comment:
Hm would it be better to not add `c.sending` when the caller asks for the delayed fn?

Reply:
Hm maybe not, I think this is ok.